src/heaps/fibonacci_heap.jl

```diff
@@ -0,0 +1,367 @@
+
+function findNode(head::MutableFibonacciHeapNode{T}, key::T) where T
+    found::Union{MutableFibonacciHeapNode,Nothing} = nothing
+    temp = head
+    temp.flag = true
+    if temp.value === key
+        found = temp; 
+        temp.flag = false; 
+        return found; 
+    end
+    if found === nothing
+        if temp.child !== nothing
+            return findNode(temp.child, key); 
+        end
+        if (temp.right).flag === false
+            return findNode(temp.right, key); 
+        end
+    end
+    temp.flag = false;
+
+end
+#################################################
+#
+#   interfaces
+#
+#################################################
+
+length(h::MutableFibonacciHeap) = h.nodeCount
+
+isempty(h::MutableFibonacciHeap) = h.nodeCount == 0
+
+# function sizehint!(h::MutableFibonacciHeap, s::Integer)
+#     sizehint!(h.nodes, s)
+#     sizehint!(h.node_map, s)
+#     return h
+# end
+
+"""
+        top(h::MutableFibonacciHeap)
+
+Returns the value of top element in heap `h`
+"""
+@inline top(h::MutableFibonacciHeap) = h.minNode.value
+
+"""
+        top_with_handle(h::MutableFibonacciHeap)
+
+Returns the top element of the heap `h` and its handle.
+"""
+@inline minimum(h::MutableFibonacciHeap) = h.minNode.value
+
+@inline top_with_handle(h::MutableFibonacciHeap) = h.minNode
+
+@inline minimum_with_handle(h::MutableFibonacciHeap) = h.minNode
+"""
+    push!{T}(h::MutableFibonacciHeap{T},data::T)
+
+Insert element into the heap `h` and return its handle.
+"""
+function push!(root::MutableFibonacciHeap,data::T) where T
+    node = MutableFibonacciHeapNode(data)
+    node.left = node.right = node
+    _merge_with_root_list(root, node)
+    if root.minNode === nothing || node.value < root.minNode.value
+        root.minNode = node
+    end
+    root.nodeCount += 1
+    return node
+end
+
+"""
+    pop!(h::MutableFibonacciHeap)
+
+Delete the top element node from heap `h` and return its handle
+"""
+pop!(root::MutableFibonacciHeap) = extractMin(root)
+
+popmin!(root::MutableFibonacciHeap) = extractMin(root)
+"""
+    merge!{T}(h1::MutableFibonacciHeap{T}, h2::MutableFibonacciHeap{T})
+
+merge two fibonacci heapsby concatenating the root lists of heap `h1` and `h2`
+returns handler to merged list
+"""
+function merge!(h1::MutableFibonacciHeap{T}, h2::MutableFibonacciHeap{T}) where T
+    newh = FibonacciHeap{T}()
+    newh.rootList, newh.minNode = h1.rootList, h1.minNode
+    last = h2.rootList.left
+    h2.rootList.left = newh.rootList.left
+    newh.rootList.left.right = h2.rootList
+    newh.rootList.left = last
+    newh.rootList.left.right = newh.rootList
+    if h2.minNode.value < newh.minNode.value
+        newh.minNode = h2.minNode
+    end
+    newh.nodeCount = h1.nodeCount + h2.nodeCount
+    return newh
+end
+
+"""
+    update!{T}(h::MutableFibonacciHeap{T}, x::Any, k::Any}
+
+modify the data of node `x` in the heap `h` to value `k`
+"""
+function update!(root::MutableFibonacciHeap{T}, oldkey::T, newkey::T) where T
```

Comment:
Same as above, and in both this function and `delete!`, the signature of the documentation do not match the signature of the function.